src/wms/src/test/java/org/geoserver/wms/map/RenderedImageMapOutputFormatTest.java

```diff
@@ -691,6 +692,81 @@ public void testDirectVsNonDirectRasterRender() throws Exception {
         imageMap.dispose();
     }
 
+    @Test
+    public void testDirectRasterRenderRespectingFormatOptions() throws Exception {
+        Catalog catalog = getCatalog();
+        CoverageInfo ci =
+                catalog.getCoverageByName(
+                        SystemTestData.WORLD.getPrefix(), SystemTestData.WORLD.getLocalPart());
+
+        CoordinateReferenceSystem crs = DefaultGeographicCRS.WGS84;
+
+        // Request an area going over the dateline
+        ReferencedEnvelope bbox =
+                new ReferencedEnvelope(new Envelope(115.0, 315.0, -60.0, 60.0), crs);
+
+        GetMapRequest request = new GetMapRequest();
+        request.setBbox(bbox);
+        request.setSRS("urn:x-ogc:def:crs:EPSG:4326");
+        request.setFormat("image/png");
+
+        testWithFormatOptions(ci, request, bbox, false);
+        testWithFormatOptions(ci, request, bbox, true);
+    }
+
+    private void testWithFormatOptions(
+            CoverageInfo ci,
+            GetMapRequest request,
+            ReferencedEnvelope bbox,
+            boolean formatOptionsEnabled)
+            throws IOException {
+        request.getFormatOptions().put("mapWrapping", Boolean.toString(formatOptionsEnabled));
+        request.getFormatOptions()
+                .put("advancedProjectionHandling", Boolean.toString(formatOptionsEnabled));
+
+        final WMSMapContent map = new WMSMapContent(request);
+        final float width = 400;
+        map.setMapWidth((int) width);
+        map.setMapHeight(240);
+        map.setBgColor(Color.red);
+        map.setTransparent(false);
+        map.getViewport().setBounds(bbox);
+
+        StyleBuilder builder = new StyleBuilder();
+        GridCoverage2DReader reader = (GridCoverage2DReader) ci.getGridCoverageReader(null, null);
+        reader.getCoordinateReferenceSystem();
+        Layer l =
+                new CachedGridReaderLayer(
+                        reader, builder.createStyle(builder.createRasterSymbolizer()));
+        map.addLayer(l);
+
+        RenderedImageMap imageMap = this.rasterMapProducer.produceMap(map);
+        RenderedImage image = imageMap.getImage();
+        float xOffset = 150;
+        ImageWorker worker =
+                new ImageWorker(image)
+                        .crop(xOffset, 0, width - xOffset, 240)
+                        .translate(-xOffset, 0, null);
+
+        if (formatOptionsEnabled) {
+            // We have enabled continuous map wrapping.
+            // So cropping over the dateline we will see again
+            // North and South America
+            ImageAssert.assertEquals(
+                    new File(
+                            "src/test/resources/org/geoserver/wms/map/direct-raster-format-options-expected.tif"),
```

Comment:
Seems like a simple image, why use a TIFF for it rather than a PNG? I don't see exotic bands or data types being used.

Reply:
Right. I have copied it from an existing test. It was actually a PNG.
I have also updated the existing test, renaming the image to PNG.